src/cmd/turnkey/pkg/wallets.go

```diff
@@ -0,0 +1,350 @@
+		}
+
+		if !resp.IsSuccess() {
+			OutputError(eris.Errorf("failed to create wallet account: %s", resp.Error()))
+		}
+
+		Output(resp.Payload)
+	},
+}
+
+var walletAccountsListCmd = &cobra.Command{
+	Use:   "list",
+	Short: "Return accounts for the wallet",
+	PreRun: func(cmd *cobra.Command, args []string) {
+		if walletName == "" {
+			OutputError(eris.New("name for wallet must be specified"))
+		}
+	},
+	Run: func(cmd *cobra.Command, args []string) {
+		wallet, err := lookupWalletByName(walletName)
+		if err != nil {
+			OutputError(eris.Wrap(err, "failed to lookup wallet"))
+		}
+
+		params := wallets.NewGetWalletAccountsParams()
+
+		params.SetBody(&models.GetWalletAccountsRequest{
+			OrganizationID: &Organization,
+			WalletID:       wallet.WalletID,
+		})
+
+		if err := params.Body.Validate(nil); err != nil {
+			OutputError(eris.Wrap(err, "request validation failed"))
+		}
+
+		resp, err := APIClient.V0().Wallets.GetWalletAccounts(params, APIClient.Authenticator)
+		if err != nil {
+			OutputError(eris.Wrap(err, "request failed"))
+		}
+
+		if !resp.IsSuccess() {
+			OutputError(eris.Errorf("failed to list wallets: %s", resp.Error()))
+		}
+
+		Output(resp.Payload)
+	},
+}
+
+func lookupWalletByName(name string) (*models.Wallet, error) {
+	params := wallets.NewGetWalletsParams()
+
+	params.SetBody(&models.GetWalletsRequest{
+		OrganizationID: &Organization,
+	})
+
+	if err := params.Body.Validate(nil); err != nil {
+		OutputError(eris.Wrap(err, "request validation failed"))
+	}
+
+	resp, err := APIClient.V0().Wallets.GetWallets(params, APIClient.Authenticator)
+	if err != nil {
+		OutputError(eris.Wrap(err, "request failed"))
+	}
+
+	if !resp.IsSuccess() {
+		OutputError(eris.Errorf("failed to list wallets: %s", resp.Error()))
+	}
+
+	for _, wallet := range resp.Payload.Wallets {
+		if *wallet.WalletName == name {
+			return wallet, nil
+		}
+	}
+
+	return nil, eris.Errorf("wallet name %q not found in list of wallets", name)
+}
+
+func listAccountsForWallet(walletID *string) ([]*models.WalletAccount, error) {
+	params := wallets.NewGetWalletAccountsParams()
+
+	params.SetBody(&models.GetWalletAccountsRequest{
+		OrganizationID: &Organization,
+		WalletID:       walletID,
+	})
+
+	if err := params.Body.Validate(nil); err != nil {
+		OutputError(eris.Wrap(err, "request validation failed"))
+	}
+
+	resp, err := APIClient.V0().Wallets.GetWalletAccounts(params, APIClient.Authenticator)
+	if err != nil {
+		OutputError(eris.Wrap(err, "request failed"))
+	}
+
+	if !resp.IsSuccess() {
+		OutputError(eris.Errorf("failed to list wallets: %s", resp.Error()))
+	}
+
+	return resp.Payload.Accounts, nil
+}
+
+func getCurveForAddressFormat(addressFormat models.AddressFormat) *models.Curve {
+	switch addressFormat {
+	case models.AddressFormatEthereum, models.AddressFormatCosmos, models.AddressFormatUncompressed:
+		return models.NewCurve(models.CurveSecp256k1)
+	case models.AddressFormatSolana:
+		return models.NewCurve(models.CurveEd25519)
+	default:
```

Comment:
i just realized that we now have this logic in multiple places for predicting address format, path format, and path. what do you think about actually just modifying the wallet account intent to not require those 3 fields and perform this logic in one place on the backend? i guess the risk is if our assumptions about these defaults no longer hold, but in either case they can set it themselves by passing these 3 fields themselves and we could document it. (anyway, not a discussion for this particular PR!)

Reply:
IMO it's best if we allow the activity itself to be super flexible. That'll end up being more important once we've allowed import since we can't assume the paths that were used to create their accounts. We could definitely have a route that helps generate the payload to sign for you, but the interface for the UI was slightly different than the interface for the CLI... so 🤷 

These shouldn't need to be updated too frequently so I think it's probably fine.

Reply:
SGTM. def agree that this flexibility is a feature especially for import.